lib/ts/recipe/passwordless/api/consumeCode.ts

```diff
@@ -56,24 +57,43 @@ export default async function consumeCode(apiImplementation: APIInterface, optio
         });
     }
 
+    const userContext = makeDefaultUserContextFromAPI(options.req);
     let result = await apiImplementation.consumeCodePOST(
         deviceId !== undefined
             ? {
                   deviceId,
                   userInputCode,
                   preAuthSessionId,
                   options,
-                  userContext: makeDefaultUserContextFromAPI(options.req),
+                  userContext,
               }
             : {
                   linkCode,
                   options,
                   preAuthSessionId,
-                  userContext: makeDefaultUserContextFromAPI(options.req),
+                  userContext,
               }
     );
 
     if (result.status === "OK") {
+        if (result.user.email !== undefined) {
+            const emailVerificationInstance = EmailVerification.getInstance();
+            if (emailVerificationInstance) {
+                const tokenResponse = await emailVerificationInstance.recipeInterfaceImpl.createEmailVerificationToken({
+                    userId: result.user.id,
+                    email: result.user.email,
+                    userContext,
+                });
+
+                if (tokenResponse.status === "OK") {
+                    await emailVerificationInstance.recipeInterfaceImpl.verifyEmailUsingToken({
+                        token: tokenResponse.token,
+                        userContext,
+                    });
+                }
+            }
+        }
```

Comment:
This should probably happen inside the implementation file. Before we create a new session? Like how we do here: https://github.com/supertokens/supertokens-node/blob/a13fd982f6fc72d3b804b3ddc70d02d82bfc6879/lib/ts/recipe/thirdparty/api/implementation.ts#L159

It is important to do it before creating a session cause then the session's emailverification claim will get that the email is verified, otherwise it won't get that